econml/grf/_base_grf.py

```diff
@@ -793,10 +793,13 @@ def predict_full(self, X, interval=False, alpha=0.05):
         """
         if interval:
             point, pred_var = self._predict_point_and_var(X, full=True, point=True, var=True)
+            assert np.isclose(pred_var[pred_var < 0], 0, atol=1e-8).all(), '`pred_var` must be > 0'

```

Comment:
pred_var is a covariance matrix of dimension (n_samples, n_outputs_, n_outputs_)

only the diagonal elemens, i.e. pred_var[:, t, t] for some t need to be non-negative. 

So make sure the assertion is only on those elements and that the maximum is only applied to those diagonal elements.

Reply:
kk I'l; confirm the dimensionality of the output there and change the assertion